WordPress/src/main/java/org/wordpress/android/util/LocaleManager.java

```diff
@@ -168,12 +150,15 @@ public static Locale languageLocale(@Nullable String languageCode) {
         if (TextUtils.isEmpty(languageCode)) {
             return Locale.getDefault();
         }
-
-        if (languageCode.length() > NO_REGION_LANG_CODE_LEN) {
-            return new Locale(languageCode.substring(0, NO_REGION_LANG_CODE_LEN),
-                    languageCode.substring(REGION_SUBSTRING_INDEX));
+        if (languageCode.contains("_")) {
+            // Attempt to parse language and region codes.
+            String[] opts = languageCode.split("_");
```

Comment:
I found this code called so many times during app execution. What are your thoughts about using a pre-compiled regexp to split the strings?
`private static Pattern languageSplitter = Pattern.compile("_");` and then use it with `String[] opts = languageSplitter.split(languageCode, 0);`

Do you think that this will improve performance?

Reply:
I tested this theory out in a little side app and indeed, the pattern is faster. It takes a lot of calls (over 500), but still an optimization I see as being worthy. I'll update the code. Thanks @daniloercoli !